server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -689,4 +713,17 @@ public void onFailure(Exception e) {
             )
         );
     }
+
+    /**
+     * TODO: should I use Stage or some other state? It seems the most precise for having PAUSING etc, but look around.
+     */
+    public Map<Stage, Long> getCountOfInProgressShardSnapshotsByShardSnapshotStage() {
```

Comment:
Changed to actively tracking shard snapshots, rather than counting them on demand. Keeping this around for discussion, but plan to remove it.